### ISSUE TITLE
Assert in SRE during Boo compiler run with Mono 6.x

### ISSUE THREAD

**Issue by tpokorra** [boo-lang/boo#209]
I am the package maintainer of boo in Fedora.
For a while, there is an issue that the build does not work anymore, with Mono 6.x.
This is the bug in Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=1799197

I have reported it upstream: https://github.com/mono/mono/issues/18970 but that did not cause any action (yet).

I can reduce it to these steps:

```
cd /root/rpmbuild/BUILD/boo-unstable
nant -D:install.prefix=/usr -D:docs.dir=/usr/share/doc -D:install.booexamples=/usr/share/boo/examples
mono build/booc.exe -v -noconfig -debug -out:build/Boo.NAnt.Tasks.dll -target:library -r:NAnt.Core.dll -r:NAnt.DotNetTasks.dll -srcdir:src/Boo.NAnt.Tasks -lib:/usr/lib/NAnt/bin/
```

Start of the output:

```
Boo Compiler version 0.9.7.0 (6.8.0.123 (tarball Sat Jun 27 04:51:00 UTC 2020))
* Assertion at sre-encode.c:290, condition `count > 0' not met
```

see attached files for details.
[boo_build_failure.txt](https://github.com/boo-lang/boo/files/5078278/boo_build_failure.txt)

Is there anything that could be done from the side of boo?

**Comment by tpokorra** [comment]
it looks quite similar to https://xamarin.github.io/bugzilla-archives/45/45136/bug.html but is probably different.

**Comment by masonwheeler** [comment]
Yeah, from the build log, that really does look like a project in the Mono runtime.
